WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/history/ScanHistoryListViewModel.kt

```diff
@@ -76,8 +78,18 @@ class ScanHistoryListViewModel @Inject constructor(
     private fun filterByTabType(threatList: List<ThreatModel>, tabType: ScanHistoryTabType) =
             threatList.filter { mapTabTypeToThreatStatuses(tabType).contains(it.baseThreatModel.status) }
 
-    private fun mapToThreatUiStateList(threatList: List<ThreatModel>) =
-            threatList.map { scanThreatItemBuilder.buildThreatItem(it, this::onItemClicked) }
+    private fun mapToThreatUiStateList(threatList: List<ThreatModel>): List<ScanListItemState> {
+        val uiStates = mutableListOf<ScanListItemState>()
+        threatList.forEach { model ->
+            val currentItem = scanThreatItemBuilder.buildThreatItem(model, this::onItemClicked)
+            val previousItem = uiStates.lastOrNull() as? ThreatItemState
+            if (previousItem == null || currentItem.firstDetectAt != previousItem.firstDetectAt) {
+                uiStates.add(ThreatDateItemState(currentItem.firstDetectAt))
+            }
+            uiStates.add(currentItem)
+        }
+        return uiStates
+    }
```

Comment:
Question ❓ : How do you feel about splitting `mapToThreatUiStateList` into two parts
- one that actually maps to threat item state
- another that adds date labels?

At first glance, it might not be clear enough that this mapping also adds date headers, wdyt?

Reply:
I've separated the adding date header to different function.